packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -80,12 +80,37 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
 
     async setRoot(uri: string): Promise<void> {
         this.root = new Deferred();
+        const listUri: string[] = [];
+        const oldListUri = await this.getRecentRoots();
+        listUri.push(uri);
+        if (oldListUri) {
+            oldListUri.forEach(element => {
+                // Store the new recentRoots as the first one.
+                // Remove duplicate
```

Comment:
Are these comments valid?
 - *Store the new recentRoots as the first one.*: You already pushed the `uri` as the first item.
 - *Remove duplicate*: you are not checking the duplicates, right?

Reply:
OK comment removed, was from my initial version